toolkit/tools/imagegen/configuration/systemconfig.go

```diff
@@ -61,12 +73,35 @@ func (s *SystemConfig) IsValid() (err error) {
 				return fmt.Errorf("empty kernel entry found in the [KernelOptions] field (%s); remember that kernels are FORBIDDEN from appearing in any of the [PackageLists]", name)
 			}
 		}
+	}
+
+	// Validate the partitions this system config will be including
+	mountPointUsed := make(map[string]bool)
+	for _, partitionSetting := range s.PartitionSettings {
+		if err = partitionSetting.IsValid(); err != nil {
+			return fmt.Errorf("invalid [PartitionSettings]: %w", err)
+		}
+		if mountPointUsed[partitionSetting.MountPoint] {
+			return fmt.Errorf("invalid [PartitionSettings]: duplicate mount point found at '%s'", partitionSetting.MountPoint)
+		}
+		mountPointUsed[partitionSetting.MountPoint] = true
+	}
+
+	if s.ReadOnlyVerityRoot.Enable || s.Encryption.Enable {
+		if !mountPointUsed["/"] {
+			return fmt.Errorf("invalid [ReadOnlyVerityRoot] or [Encryption]: must have a partition mounted at '/'")
+		}
+		if s.ReadOnlyVerityRoot.Enable && s.Encryption.Enable {
+			return fmt.Errorf("invalid [ReadOnlyVerityRoot] & [Encryption]: verity root currently does not support root encryption")
```

Comment:
>& [](start = 51, length = 1)

Nit: for the sake of consistency let's use "and" instead of "&".

Reply:
Fixed